File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
  let mapper = new Map();

    
  for (let i = 0; i< nums.length; i++){
      const currentValue = nums[i]
      if (mapper.has(currentValue) && i - mapper.get(currentValue) <= k){
          return true
      }
      else{
          mapper.set(currentValue, i)
      }
  }
    return false;
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hashmap = {}
        
        for i in range(len(nums)):
            current = nums[i]
            if current in hashmap and i - hashmap[current] <= k:
                return True
            else:
                hashmap[current] = i
                
        
        return False